vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -361,8 +371,51 @@ default T getOrElse(Supplier<? extends T> supplier) {
     }
 
     /**
-     * Maps the value and wraps it in a new {@code Some} if this is a {@code Some}, returns {@code None}.
+     * Maps the value to a new {@code Option} if this is a {@code Some}, otherwise returns {@code None}.
+     * In the case the given {@code mapper} throws, {@code None} will be returned.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * // = None
+     * Option.<Integer> none().flatMapTry(i -> 1 / i);
+     *
+     * // = Some(0)
+     * Option.some(2).flatMapTry(i -> Option.of(1 / i));
      *
+     * // = None
+     * Option.some(0).flatMapTry(i -> Option.of(1 / i)); // division by zero
+     * }</pre>
+     *
+     * @param mapper A value mapper that may throw
+     * @param <U>    Component type of the resulting Option
+     * @return a new {@code Option}
+     */
+    @SuppressWarnings("unchecked")
+    default <U> Option<U> flatMapTry(Function<? super T, ? extends Option<? extends U>> mapper) {
```

Comment:
Sorry, I forgot to drop some lines about the design decision.

Basically we want to do a map in the example (see original issue), but it does not accept a function that throws. Instead of adding new concepts to our 'language', we use the same things we already have in Try and Future: namely mapTry and flatMapTry. That's the most straight-forward way to do it.